src/storm-pomdp/modelchecker/PreprocessingPomdpValueBoundsModelChecker.cpp

```diff
@@ -0,0 +1,321 @@
+#include "PreprocessingPomdpValueBoundsModelChecker.h"
+#include <random>
+
+#include "storm-pomdp/storage/PomdpMemory.h"
+#include "storm-pomdp/transformer/PomdpMemoryUnfolder.h"
+
+#include "storm/modelchecker/results/ExplicitQuantitativeCheckResult.h"
+#include "storm/storage/Scheduler.h"
+
+#include "environment/solver/MinMaxSolverEnvironment.h"
+#include "environment/solver/TopologicalSolverEnvironment.h"
+#include "storm/exceptions/NotSupportedException.h"
+#include "storm/exceptions/UnexpectedException.h"
+#include "storm/utility/macros.h"
+#include "storm/utility/vector.h"
+
+namespace storm {
+    namespace pomdp {
+        namespace modelchecker {
+            template <typename ValueType>
+            PreprocessingPomdpValueBoundsModelChecker<ValueType>::PreprocessingPomdpValueBoundsModelChecker(storm::models::sparse::Pomdp<ValueType> const& pomdp, storm::solver::MinMaxMethod minMaxMethod) : pomdp(pomdp) {
+                // Setup checker environment
+                mcEnvironment = storm::Environment();
+                mcEnvironment.solver().setLinearEquationSolverType(storm::solver::EquationSolverType::Topological, true);
+                mcEnvironment.solver().minMax().setMethod(minMaxMethod);
+                mcEnvironment.solver().minMax().setPrecision(storm::utility::convertNumber<storm::RationalNumber>(1e-9));
+                mcEnvironment.solver().topological().setUnderlyingEquationSolverType(storm::solver::EquationSolverType::Eigen);
```

Comment:
Why do we require topological + Eigen for DTMCs and (non-topological) SVI for MDPs?
Maybe, what we actually want here is something like `setForceExact` or `setForceSoundness` in `SolverEnvironment` ?
In any case, it would be nice to elaborate a bit on this.

Usually, we pass a `storm::Environment cons& env` to all the methods that need an environment, i.e., it should be added as the first argument of `getValueBounds` and `getExtremeValueBound`.  Then, the calling method needs to take care of configuring the environment which makes things a bit more transparent.

There might be reasons to do it differently here that I don't see right now, though.